standard/attributes.md

```diff
@@ -858,6 +859,276 @@ For invocations that occur within field or event initializers, the member name u
 
 For invocations that occur within declarations of instance constructors, static constructors, finalizers and operators the member name used is implementation-dependent.
 
+### §Nullable-Analysis-Attributes Code analysis-attributes
+
+#### §Nullable-Analysis-Attributes-General General
+
+Code compiled with both nullable contexts (§8.9.4) disabled is null oblivious (§8.9.5). That means any reference type variable may be null, but null checks aren't required. Once such code is made nullable-aware, those rules change. Reference type variables should never have the null value, and such variables must be checked against null before being dereferenced.
+
+Some APIs have more complex rules for when variables can or can't be null. In these cases, one or more of the nullable-related attributes described below can be used to express those rules. When user code is compiled in a nullable-enabled context, the compiler is required to warn when that code violates those rules. That is, these attributes help define the nullability contract for an API.
```

Comment:
“Once such code is made nullable-aware, those rules change” -> “Nullable-aware code follows different rules” – or something like that. Users might write new code which is “null oblivious” and other code which is “nullable-aware” – i.e. for former need not be old and the latter need not have been the former, if that makes sense!

“The complier is required to warn…” – first a compiler is not required to have a null analysis is it? And if it does as the algorithm it uses it implementation defined is it required to enforce the semantics of these attributes or could it just ignore them?

I see the purpose of listing these attributes as (a) enforcing that they are reserved and (b) [probably?] saying that if a compiler chooses to use them it should do so with the meanings listed. Anything more than that is up to the implementation. But maybe we’re going to go further than that?

Reply:
Yes, the next commits updated the language to reflect that.

I'm leaving the examples as instructive (and one more PR is to create more examples on nullability and null states)